R/create_task_makefile.R

```diff
@@ -82,23 +83,43 @@ create_task_makefile <- function(
   }), use.names=FALSE)
   job_output <- if(isTRUE(ind_complete) || is.na(ind_complete)) {
     if(is.null(ind_dir)) stop('ind_dir must not be NULL when ind_complete=TRUE')
-    file.path(ind_dir, as_ind_file(job_target, ind_ext))
+    file.path(ind_dir, as_ind_file(target_name, ind_ext))
   } else {
-    sprintf('%s_%s', job_target, ind_ext)
+    sprintf('%s_%s', target_name, ind_ext)
   }
+  
+  # prepare the task list for rendering
+  tasks <- unname(task_plan) # remove list element names where they'd interfere with whisker.render
+  for(task in seq_along(tasks)) {
+    # remove lmore ist element names where they'd interfere with whisker.render
+    tasks[[task]]$steps <- unname(tasks[[task]]$steps)
+    for(step in seq_along(tasks[[task]]$steps)) {
+      # add a logical for has_depends
+      tasks[[task]]$steps[[step]]$has_depends <- length(tasks[[task]]$steps[[step]]$depends) > 0
+    }
+  }
+  
   job <- list(
-    target_name = job_target,
+    target_name = target_name,
     # even though target_name is an object (not file), job_command should write
     # to ind_file - again so the calling remake file can use
     # ind_file as its target
     command = {
+      if(!is.null(finalize_fun)){
+        if(isTRUE(ind_complete) | is.na(ind_complete)) {
+          message('ind_complete=', ind_complete, ' is incompatable with ', finalize_fun, '(). Using FALSE')
```

Comment:
I don't think `ind_complete` and the output of `finalize_fun` should be forced not to coexist, though I could see a warning or message encouraging the user not to bother with `ind_complete=TRUE` if they're providing one or more combiners.

Reply:
Part of this is temporary as I work through the implementation. But I am still wondering about the need to indicate the final single output within this yaml, vs sc_indicate from the parent yaml.

Reply:
Agreed that there's no need to produce an .ind file for a combiner output; a target name specific to that combiner, of the user's choosing, would be fine.

I do think we still want an option to produce the .ind file for summary output, because that's the one that can put together a single list of filenames and hashes for the final steps of all of the tasks. The target that's implemented for that summary output does write an .ind file but uses an object name / symbol as the recipe target name (for the reason noted at https://github.com/USGS-R/scipiper/issues/52#issuecomment-448623127)